hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/shortcircuit/ShortCircuitCache.java

```diff
@@ -240,7 +245,7 @@ public void run() {
         } else {
           shm.getEndpointShmManager().shutdown(shm);
           IOUtilsClient.cleanupWithLogger(LOG, domainSocket, out);
-          domainSocket = null;
```

Comment:
why do you remove the line 243?  If remove it, the line 228  looks redundant.

Reply:
> why do you remove the line 243?

`domainSocket` is now a local variable (defined at line 192). We don't need to update it at the end of this method. Note that this `finally` clause belongs to the try-clause starts at line 195.

> If remove it, the line 228 looks redundant.

For line 228, it's needed in the next retry of the while-loop. The check at line 199 will catch it and trigger a re-connect.